lisa/tools/find.py

```diff
@@ -0,0 +1,50 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from pathlib import PurePath
+from typing import List
+
+from lisa.executable import Tool
+from lisa.util import LisaException
+
+
+class Find(Tool):
+    @property
+    def command(self) -> str:
+        return "find"
+
+    @property
+    def can_install(self) -> bool:
+        return False
+
+    def find_files(
+        self,
+        start_path: PurePath,
+        name_pattern: str = "",
```

Comment:
It looks the name pattern can be a part of path pattern. Do we really need this one?

Reply:
Yes we need it, both have different purpose. For example - assume we don't know the full system paths to a specific name folder and want to get the path to that folder. So I run this 
find /sys/devices/ -iname '*net*' -print0

try running it with -ipath. It will give very different result.

Reply:
in this case, it can be "-ipath */net"

Reply:
No it won't be same. I just tried. My above comment hid the stars I put in value for -iname 
try this - 
find /sys/devices/ -iname '<star>net<star>' -print0
find /sys/devices/ -ipath '<star>/<star>net<star>' -print0

And ofcourse there are other cases too where we are not sure of the path or the name.